pkg/registration/spoke/options.go

```diff
@@ -8,15 +8,27 @@ import (
 	"github.com/spf13/pflag"
 	"k8s.io/apiserver/pkg/server/healthz"
 
+	ocmfeature "open-cluster-management.io/api/feature"
+
+	"open-cluster-management.io/ocm/pkg/features"
 	"open-cluster-management.io/ocm/pkg/registration/helpers"
 )
 
 var ClientCertHealthCheckInterval = 30 * time.Second
 
 // SpokeAgentOptions holds configuration for spoke cluster agent
 type SpokeAgentOptions struct {
-	BootstrapKubeconfig         string
-	BootstrapKubeconfigSecret   string
+	// The differences among BootstrapKubeconfig, BootstrapKubeconfigSecret, BootstrapKubeconfigSecrets are:
+	// 1. BootstrapKubeconfig is a file path, the controller uses it to build the client.
+	// 2. BootstrapKubeconfigSecret is the secret, a eventhandler will watch it, if the secret is changed, threbootstrap.
+	// 3. BootstrapKubeconfigs is a list of file path, the controller uses one of its item to build the client.
+	// BootstrapKubeconfigs can only be used when MultipleHubs is enabled.
+	BootstrapKubeconfig       string
+	BootstrapKubeconfigSecret string
+	BootstrapKubeconfigs      []string
+
+	HubConnectionTimeoutSeconds int32
```

Comment:
What this value should be set, is this related to the leaseDurationSecond in managedcluster spec?

Reply:
The value is not necessarily related to leaseDurationSecond; by default, we set the value to 10 minutes and at least 3 minutes:
```yaml
                          hubConnectionTimeoutSeconds:
                            default: 600
                            description: HubConnectionTimeoutSeconds is used to set
                              the timeout of connecting to the hub cluster. When agent
                              loses the connection to the hub over the timeout seconds,
                              the agent do a rebootstrap. By default is 10 mins.
                            format: int32
                            minimum: 180
```

Reply:
why it is not related? It will not make sense that leaseDurationSecond is set to 5 mins and hubConnectionTimeoutSeconds is 1 mins.

Reply:
If the leaseDurationSecond is 5min and hubConnnectionTimeoutSeconds is 1 min I prefer consider it as a configuration error of user.

Do you mean we should set the timeout seconds a number times the leaseDurationSecond? I'm not sure if it can suite every customer's case. For example, if the leaseDurationSecond set to 1min is a common thing, but in different cases, customer may want to set hubTimeoutSeconds to 3mins, 5mins or 10mins.

Reply:
Do you think we can add a step in the validate to check hubTimeoutSeconds must bigger than the leaseDurationSeconds?

Reply:
leaseDurationSeconds can be configured in ManagedCluster, so add such step won't work. I think we need to at least add some doc here. It is ok as the first step to have hubConnnectionTimeoutSeconds as an flag, but we should have a TODO to retrieve it from leaseDurationSeconds.

Reply:
Another reason is `leaseDurationSeconds` is a field per managedcluster. 
```go
observedLeaseDurationSeconds := cluster.Spec.LeaseDurationSeconds
```

But we set hubTimetoutSeconds as a spoke-agent configuration field. We can't compare those 2 fields in the Validate function before agent run.

Reply:
I think there is another way to ensure timeout always greater than leaseDuration, we already has a min value(180 seconds) set for hubTimeoutSeconds in api definenation: https://github.com/open-cluster-management-io/api/blob/9d94703b9eba006e618a68588000722bb945112f/operator/v1/types_klusterlet.go#L209

We can also set a max value(60 seconds) for the leaseDurationSeconds(), so that one is always greater than another?

Reply:
we cannot set maximum, one of the reason to allow configuration is in the pool network scenario, we allow user to set it longer to tolerate temp network disruption.

Reply:
Agree, the TODO added.